### PR TITLE
icub3: fixes clash of config files with icub2

### PR DESCRIPTION
**Pull request by Nicogene**
It should fix https://github.com/robotology/icub-models/issues/50

Please review code.

### CODE REVIEW

**Comment by traversaro** [comment]
As the project install all its models in `<install_prefix>/share/iCub`, I think it would be a bit strange to start install stuff also in `<install_prefix>/share/iCub3`, especially because the iCub3 models would still be installed in  `<install_prefix>/share/iCub` . Can we just install conflicting files in `conf_icub3` ? For the meshes I guess we do not even need to use a different directory, as the meshes do not have any name collision.

**Comment by traversaro** [comment]
> Can we just install conflicting files in `conf_icub3` ?

Or also in `<install_prefix>/share/iCub/conf/iCub3` , any option is fine for me as long as we install everything install `<install_prefix>/share/iCub` .

File: simmechanics/CMakeLists.txt [hunk 1]
```diff
@@ -335,5 +335,5 @@ add_custom_command(TARGET generate-models-simmechanics
                    COMMAND ${CMAKE_COMMAND} -E copy_directory "${CMAKE_CURRENT_SOURCE_DIR}/data/icub2_5/meshes" "${CMAKE_BINARY_DIR}/${BUILD_PREFIX}/meshes"
                    COMMAND ${CMAKE_COMMAND} -E copy_directory "${CMAKE_CURRENT_SOURCE_DIR}/data/icub2_5/conf" "${CMAKE_BINARY_DIR}/${BUILD_PREFIX}/conf"
                    COMMAND ${CMAKE_COMMAND} -E copy_directory "${CMAKE_CURRENT_SOURCE_DIR}/data/icub3/meshes" "${CMAKE_BINARY_DIR}/${BUILD_PREFIX}/meshes"
-                   COMMAND ${CMAKE_COMMAND} -E copy_directory "${CMAKE_CURRENT_SOURCE_DIR}/data/icub3/conf" "${CMAKE_BINARY_DIR}/${BUILD_PREFIX}/conf"
+                   COMMAND ${CMAKE_COMMAND} -E copy_directory "${CMAKE_CURRENT_SOURCE_DIR}/data/icub3/conf" "${CMAKE_BINARY_DIR}/${BUILD_PREFIX}/conf_icub3"
```
Comment (Nicogene) [inline simmechanics/CMakeLists.txt:338]: @traversaro it will install the files in `conf_icub3` right?

Comment (traversaro) [reply simmechanics/CMakeLists.txt:338]: I guess so, but I am a bit afraid that `copy_directory` could have some strange logic for source and destination directory named in the same way. The quickest way to check is probably to check it locally.

**Review by traversaro — APPROVED** [review]

**Pull request merged**